tests/test_dom0_and_templates_correct_key.py

```diff
@@ -0,0 +1,248 @@
+import json
+import os
+import subprocess
+import tempfile
+from datetime import datetime
+from string import Template
+
+import pytest
+from qubesadmin import Qubes
+
+FEDORA_VERSION = "f37"
+DEBIAN_DIST = "bookworm"
+
+REPO_CONFIG = {
+    "prod": {
+        "signing_key": "/etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation",
+        "repo_file_name": "securedrop-workstation-dom0.repo",
+        "yum_repo_url": "https://yum.securedrop.org/workstation/dom0/$FEDORA_VERSION/",
+    },
+    "dev": {
+        "signing_key": "/etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test",
+        "repo_file_name": "securedrop-workstation-dom0-dev.repo",
+        "yum_repo_url": "https://yum-test.securedrop.org/workstation/dom0/$FEDORA_VERSION-nightlies/",
+    },
+    "staging": {
+        "signing_key": "/etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test",
+        "repo_file_name": "securedrop-workstation-dom0-staging.repo",
+        "yum_repo_url": "https://yum-test.securedrop.org/workstation/dom0/$FEDORA_VERSION/",
+    },
+}
+
+
+# Get expected configuration (environment-dependent)
+def _get_expected_config():
+    with open("config.json") as c:
+        config = json.load(c)
+        env = config.get("environment")
+
+        # Fall back to checking environment via keyring package
+        # (will eventually replace config.json environment)
+        if not env:
+            env = _get_env_by_package()
+
+    config = REPO_CONFIG[env].copy()
+
+    # Fedora version is a placeholder, so fix that
+    config["yum_repo_url"] = Template(config["yum_repo_url"]).safe_substitute(
+        {"FEDORA_VERSION": FEDORA_VERSION}
+    )
+    return env, config
+
+
+def _get_env_by_package():
+    """
+    Check which environment we are using by checking keyring package.
+    Order matters; due to package versioning, an installed dev package
+    means a dev setup, so check dev, staging, then prod.
+    """
+    for env, suffix in [("dev", "-dev"), ("staging", "-staging"), ("prod", "")]:
+        if _is_installed(f"securedrop-workstation-keyring{suffix}"):
+            return env
+    pytest.fail("No keyring package")
+    # Unreachable, but ruff was unhappy
+    return None
+
+
+def _is_installed(pkg: str):
+    """
+    Check if package is installed.
+    """
+    try:
+        subprocess.check_call(
+            ["rpm", "-q", pkg], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
+        )
+
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+# A bit clunky, but we have limited tooling options in dom0.
```

Comment:
Cursed but if it works, it works :) 

Could you add an explicit `TODO: use sequoia here instead` - I think we can do that once we're on 4.3.